backend/internal/service/app_config_service.go

```diff
@@ -17,12 +18,14 @@ import (
 type AppConfigService struct {
 	DbConfig *model.AppConfig
 	db       *gorm.DB
+	mu       sync.RWMutex // Add a mutex
 }
 
 func NewAppConfigService(db *gorm.DB) *AppConfigService {
 	service := &AppConfigService{
 		DbConfig: &defaultDbConfig,
 		db:       db,
+		mu:       sync.RWMutex{}, // Initialize the mutex
```

Comment:
```suggestion
		DbConfig: &atomic.Pointer[model.AppConfig]{}
		db:       db,
```

Then set `s.DbConfig.Store(defaultDbConfig)`